code/game/objects/items/weapons/flamethrower.dm

```diff
@@ -15,239 +152,216 @@
 	if(ptank)
 		add_overlay("+ptank")
 	if(lit)
-		add_overlay("+lit")
 		item_state = "flamethrower_1"
+		icon_state = "flamethrower1"
 	else
 		item_state = "flamethrower_0"
+		icon_state = "flamethrower0"
 	if(ismob(loc))
 		var/mob/M = loc
 		M.update_inv_l_hand()
 		M.update_inv_r_hand()
 
-/obj/item/flamethrower/afterattack(atom/target, mob/user, flag)
+/obj/item/flamethrower/basic/afterattack(atom/target, mob/user, flag)
 	. = ..()
 	if(flag)
 		return // too close
-	if(user && user.get_active_hand() == src) // Make sure our user is still holding us
-		var/turf/target_turf = get_turf(target)
-		if(target_turf)
-			var/turflist = getline(user, target_turf)
-			add_attack_logs(user, target, "Flamethrowered at [target.x],[target.y],[target.z]")
-			flame_turf(turflist)
-
-/obj/item/flamethrower/attackby(obj/item/I, mob/user, params)
-	if(isigniter(I))
-		var/obj/item/assembly/igniter/IG = I
-		if(IG.secured)
-			return
-		if(igniter)
-			return
-		if(!user.drop_item())
-			return
-		IG.forceMove(src)
-		igniter = IG
-		update_icon()
-		return
-
-	else if(istype(I, /obj/item/tank/plasma))
-		if(ptank)
-			if(user.drop_item())
-				I.forceMove(src)
-				ptank.forceMove(get_turf(src))
-				ptank = I
-				to_chat(user, "<span class='notice'>You swap the plasma tank in [src]!</span>")
-			return
-		if(!user.drop_item())
-			return
-		I.forceMove(src)
-		ptank = I
-		update_icon()
-		return
-
-	else if(istype(I, /obj/item/analyzer) && ptank)
-		atmosanalyzer_scan(ptank.air_contents, user)
-	else
-		return ..()
+	startfireline(target, user)
 
-/obj/item/flamethrower/wrench_act(mob/user, obj/item/I)
-	if(status)
-		return
-	. = TRUE
+/obj/item/flamethrower/basic/wrench_act(mob/user, obj/item/I)
 	if(!I.use_tool(src, user, 0, volume = I.tool_volume))
 		return
 	var/turf/T = get_turf(src)
 	if(weldtool)
 		weldtool.forceMove(T)
 		weldtool = null
-	if(igniter)
-		igniter.forceMove(T)
-		igniter = null
 	if(ptank)
 		ptank.forceMove(T)
 		ptank = null
 	new /obj/item/stack/rods(T)
+	new /obj/item/assembly/igniter(T) //why did it care about this in the first place?
 	qdel(src)
 
-/obj/item/flamethrower/screwdriver_act(mob/user, obj/item/I)
-	if(!igniter || lit)
-		return
-	. = TRUE
-	if(!I.use_tool(src, user, 0, volume = I.tool_volume))
-		return
-	status = !status
-	to_chat(user, "<span class='notice'>[igniter] is now [status ? "secured" : "unsecured"]!</span>")
+/obj/item/flamethrower/basic/CheckParts(list/parts_list)
+	..()
+	weldtool = locate(/obj/item/weldingtool) in contents
 	update_icon()
 
-/obj/item/flamethrower/attack_self(mob/user)
-	toggle_igniter(user)
-
-/obj/item/flamethrower/AltClick(mob/user)
-	if(ptank && isliving(user) && user.Adjacent(src))
-		user.put_in_hands(ptank)
-		ptank = null
-		to_chat(user, "<span class='notice'>You remove the plasma tank from [src]!</span>")
+/obj/item/flamethrower/basic/Initialize(mapload)
+	. = ..()
+	if(create_full)
+		if(!weldtool)
+			weldtool = new /obj/item/weldingtool(src)
+		if(create_with_tank)
```

Comment:
Why have this here and in the advanced/Init? Why not have it on the flamethrower/init?